File: runtime/src/main/java/scooterdelta/challenge/bot/module/ChallengeComponent.java
package scooterdelta.challenge.bot.module;

import dagger.Component;
import scooterdelta.challenge.bot.ChallengeApplication;
import scooterdelta.challenge.bot.process.module.ProcessModule;

@Component(
        modules = {
                ProcessModule.class
        }
)
public interface ChallengeComponent {

    ChallengeApplication getChallengeApplication();

}

File: runtime/src/main/kotlin/scooterdelta/challenge/bot/component/ChallengeComponent.kt
package scooterdelta.challenge.bot.component

import dagger.Component
import scooterdelta.challenge.bot.process.module.ProcessModule
import scooterdelta.challenge.bot.runner.ChallengeRunner

@Component(
        modules = arrayOf(ProcessModule::class)
)
interface ChallengeComponent {

    fun getChallengeRunner() : ChallengeRunner

}
